mdtraj/geometry/distance.py

```diff
@@ -86,6 +85,61 @@ def compute_distances(traj, atom_pairs, periodic=True, opt=True):
         return _distance(xyz, pairs)
 
 
+def compute_distances_t(traj, atom_pairs, time_pairs, periodic=True, opt=True):
+    """Compute the distances between pairs of atoms from times 0 to t.
+
+    Parameters
+    ----------
+    traj : Trajectory
+        An mtraj trajectory.
+    atom_pairs : np.ndarray, shape=(num_pairs, 2), dtype=int
+        Each row gives the indices of two atoms involved in the interaction.
+    time_pairs : nd.array
+        Each row gives the indices of two frames.
+    periodic : bool, default=True
+        If `periodic` is True and the trajectory contains unitcell
+        information, we will compute distances under the minimum image
+        convention.
+    opt : bool, default=True
+        Use an optimized native library to calculate distances. Our optimized
+        SSE minimum image convention calculation implementation is over 1000x
+        faster than the naive numpy implementation.
+
+    Returns
+    -------
+    distances : np.ndarray, shape=(n_frames, num_pairs), dtype=float
+        The distance between each pair of atoms at t=0 and t=t.
+    """
+    xyz = ensure_type(traj.xyz, dtype=np.float32, ndim=3, name='traj.xyz', shape=(None, None, 3), warn_on_cast=False)
+    pairs = ensure_type(atom_pairs, dtype=np.int32, ndim=2, name='atom_pairs', shape=(None, 2), warn_on_cast=False)
+    times = ensure_type(time_pairs, dtype=np.int32, ndim=2, name='time_pairs', shape=(None, 2), warn_on_cast=False)
+    if not np.all(np.logical_and(pairs < traj.n_atoms, pairs >= 0)):
```

Comment:
This is a sanity check that the atom_pairs are logical indices. Do you also want to add a sanity check that all of the time_pairs indices are >= 0 and < len(traj)?

Reply:
Good thought, I added this sanity check.  I also added a unit test to check that these exceptions are correctly raised.